crates/librqbit/src/watch.rs

```diff
@@ -109,23 +121,37 @@ fn watch_thread(
                 return Ok(());
             }
         }
-        for path in ev.paths {
-            if path.extension().and_then(|e| e.to_str()) != Some("torrent") {
-                trace!(?path, "ignoring path");
-                continue;
-            }
-            let add = match read_and_validate_torrent(&path) {
-                Ok(add) => add,
-                Err(e) => {
-                    warn!(?path, "error validating torrent: {e:#}");
-                    continue;
+
+        let _results = ev
+            .paths
+            .iter()
+            .filter_map(|path| match path.extension() {
+                Some(x) if x.to_str() == Some("torrent") => match read_and_validate_torrent(path) {
+                    Ok(add_torrent) => Some((add_torrent, path)),
+                    Err(e) => {
+                        warn!(?path, "torrent file invalid: {e:#}");
+                        None
+                    }
+                },
+                Some(x) if x.to_str() == Some("magnet") => match read_and_validate_magnet(path) {
+                    Ok(add_torrent) => Some((add_torrent, path)),
+                    Err(e) => {
+                        warn!(?path, "magnet file invalid: {e:#}");
+                        None
+                    }
+                },
+                Some(_) | None => {
+                    trace!(?path, "ignoring path");
+                    None
+                }
+            })
+            .map(|(add_torrent, path)| {
```

Comment:
Instead of ```let _results = map().collect()``` (which allocates needlessly and creates an unused variable) you can use ether a for loop or [for_each](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each)

Reply:
makes sense, done